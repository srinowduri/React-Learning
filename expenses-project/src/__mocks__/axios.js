module.exports = {
   
    get: (url) => {
        if (url == 'http://localhost:2000/api/expenses'){
            return Promise.resolve({
                data: [
                    { date: '1', store: 'A', price: 11, comments: 'll'},
                    { date: '2', store: 'B', price: 22, comments: 'kk'},
                    { date: '3', store: 'C', price: 33, comments: 'mm'}
                ]
            });
        }
        else {
            return Promise.resolve({
                data: { date: '1', store: 'A', price: 11, comments: 'll'}
            })
        }
    },

    post: (url) => {
        return Promise.resolve({
            text: "Successfully Added"
        });
    },

    
    put: (url) => {
        return Promise.resolve({
            text: "Successfully Updated"
        });
    },

    delete: (url) => {
        return Promise.resolve({
            text: "Successfully Deleted"
        });
    }
}
