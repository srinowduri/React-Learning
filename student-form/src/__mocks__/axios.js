'use strict';
module.exports = {
  get: (url) => {
    return Promise.resolve({
      data: [
        {
          id: 0,
          name: 'iioog'
        },
        {
          id: 1,
          name: 'bjdf'
        }
      ]
    });
  },
  post: (url) => {
    return Promise.resolve({
      data: [
        {
          id: 0,
          name: 'ABC',
          level: 'Basics'
        }
      ]
    })
  }
}

