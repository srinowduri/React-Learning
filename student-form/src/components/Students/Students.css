.student_Table {
    text-align: left;
    table-layout: fixed;
    border-collapse: collapse;
    margin-left: -50%; 
    width: 60%;
    height: 40%;
    margin-top: 60px;
    background-color: lavender;
    float: left;
    position: relative;
}


table,td, th{
    border: 2px solid navy;
    height: 50px;
}
td{
    font-weight: bold;
    color: black;
    text-align: center;
    font-size: medium;
}
th{
    text-align: center;
    color: navy;
    font-size: large;
    font-weight: bold;
    background-color: lightblue;
}
 h1 {
     float: left;
     margin-left: -50%;
 }

 .tableButton{
    padding: 5px;
    border-radius: 5px;
    outline: none;
    background-color: rgb(23, 143, 223);
    font-size: large;
    font-weight: normal;
    cursor: pointer;
 }

 /* .student_Table tr:nth-child(even){background-color: #f2f2f2;} */

.student_Table tr:hover {background-color: #ddd;}

.student_Table th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: center;
  color: white;
}

.student_Table thead{
    background-color: aquamarine;
}

.Add {
    float: left;
    position: relative;
    margin-left: -60%;
    margin-top: 30%;

}