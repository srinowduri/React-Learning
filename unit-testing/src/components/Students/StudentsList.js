import React, { Component } from 'react';
// import AddStudentForm from '../../Practice';

class StudentsList extends Component {
        state = {
            isActive: false
        }
    
    showAndHide = () => {
        this.setState({
            isActive: !this.state.isActive
        });
        // console.log(this.state.isActive);
    }
    render () {
        return (
            <div>
                <button onClick={this.showAndHide}>          
                    {this.state.isActive ? "Register" : "Cancle"}
                </button>
                {/* {this.state.isActive ? null : <AddStudentForm />} */}
            </div>
        );
    }
}

export default StudentsList;