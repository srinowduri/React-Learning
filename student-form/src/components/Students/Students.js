import React, { Component } from 'react';
import { withRouter } from 'react-router';
import axios from 'axios';
import StudentForm from '../Add-Student/StudentForm';
import './Students.css';
class Students extends Component {
    constructor(props) {
        super(props) 
        this.state = { 
            students: [],
            studentInfo: {},
            toggleStudentForm: false
        }
     }
     componentDidMount() {
        axios.get('http://localhost:2000/api/students')
            .then(responce =>{
                const studentsFromDatabase = responce.data;
                this.setState({
                    students: studentsFromDatabase
                });
            })
     }

    renderTableData() {
        return this.state.students.map((student) => {
           const {id, name, level, emailId, phoneNum} = student 
           return (
              <tr key={id}>
                 <td>{name}</td>
                 <td>{level}</td>
                 <td>{emailId}</td>
                 <td>{phoneNum}</td>
                 <td>
                     <button className="tableButton" 
                            type="submit" 
                            onClick={this.toggleEditStudentForm.bind(this, student.id)}> Edit </button>   /

                     <button className="tableButton" type="submit">Delete</button>
                 </td>
              </tr>
           )
        })
     }

    switchStudentForm = () => {
        this.setState({
            toggleStudentForm: !this.state.toggleStudentForm,
        })
        
    }

    toggleEditStudentForm = (id, event) => {
        event.preventDefault();
        axios.get('http://localhost:2000/api/students/' + id)
         .then(response => {
            const selectedStudent = response.data;
             this.setState({
                studentInfo : selectedStudent,
                toggleStudentForm: !this.state.toggleStudentForm
             })
         })
    }
   display = isShow => {
    this.setState({ 
        toggleStudentForm: isShow
    })
   }

    render() {
 
        return (
            <div>{!this.state.toggleStudentForm ? null : <StudentForm std={this.state.studentInfo}></StudentForm>}
                
                <h1>Student List: </h1>
                <table className="student_Table">
                    <thead>
                        <tr>
                            <td>Name</td>
                            <td>Level</td>
                            <td>Email-Id</td>
                            <td>Phone</td>
                            <td>Actions</td>
                        </tr>
                    </thead>
                    <tbody>
                        {this.renderTableData()}
                    </tbody>
                </table>
                <button className="Add" type="submit" onClick={this.switchStudentForm}>Add</button>
            </div>
        )
    }
}

export default withRouter(Students);