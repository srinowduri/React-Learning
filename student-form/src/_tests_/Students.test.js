import React from 'react';
import {render, cleanup} from '@testing-library/react';
import { create } from 'react-test-renderer';
import axios from 'axios';
import Students from '../components/Students/Students';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

jest.mock('axios');

describe('show and hide student form', () => {
    it('toggle student form', () => {

        let container = create(<Students />);
        let instance = container.getInstance(0);
        expect(instance.state.toggleStudentForm).toBe(false);

        instance.switchStudentForm();
        expect(instance.state.toggleStudentForm).toBe(true);

        expect(container.toJSON()).toMatchSnapshot();
    });

    it('should fetch a list of tasks', () => {
        const getSpy = jest.spyOn(axios, 'get');
        const toDoListInstance = shallow(
          <Students />
        );
        expect(getSpy).toBeCalled();
      });

    it('should fetch list of students', () => {
        let component = create(<Students />);
        let instance = component.getInstance();
        console.log(instance.state.students);
     
        axiosMock.get.mockResolvedValueOnce({
            data: {}
        })
        
        expect(axiosMock.get).toHaveBeenCalledTimes(1);
    })
})