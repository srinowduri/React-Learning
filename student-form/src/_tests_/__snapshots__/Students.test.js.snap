// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show and hide student form toggle student form 1`] = `
<div>
  <div>
    <div
      className="form"
    >
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            htmlFor="name"
          >
            Name:
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="level"
          >
            Level:
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="email"
          >
            Email-Id:
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            htmlFor="phone"
          >
            Phone Number:
          </label>
          <input
            className="input"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
  <h1>
    Student List: 
  </h1>
  <table
    className="student_Table"
  >
    <thead>
      <tr>
        <td>
          Name
        </td>
        <td>
          Level
        </td>
        <td>
          Email-Id
        </td>
        <td>
          Phone
        </td>
        <td>
          Actions
        </td>
      </tr>
    </thead>
    <tbody />
  </table>
  <button
    className="Add"
    onClick={[Function]}
    type="submit"
  >
    Add
  </button>
</div>
`;
